### PR TITLE
Thank you for completing the tests.

### PR DESCRIPTION
**Pull request by yuzurihaaa**
Hi, I am Yusuf, one of the interviewer.

First of all, thank you for completing this task. Here's some of my feedback on your codes do review some of my codes if they are not up to your standard.

Thank you.

### CODE REVIEW

**Comment by yuzurihaaa** [comment]
Comments:
1. If possible maybe don't use deprecated libraries.
~Time~ date = new ~Time~()

2. Calendar typically have their own formatter, use it well.

3. Texts should be in `strings.xml` but maybe this is on purpose.

4. Please try not to duplicate code as much as possible.

**Comment by zulfiqarlaili** [comment]
Thanks for the clean up yusuf, i like where i can get different perspective to right the code. You really did clean up my code. haha. i like the way you make it simple to look at especially on my onclick method. its quite messy there. 

i would like to learn how to write a clean code from you in a future. Maybe is should make my other repo to public so i can get different perspective.

thank you for your time to look at my code

File: app/src/main/java/com/maybanktest/AddNewItemActivity.java [hunk 1]
```diff
@@ -43,95 +41,95 @@ private void ini() {
...
+            if (isNew) {
+                addData(taskModel);
+            } else {
+                updateData(taskModel);
+            }
+        } else {
+            Toast.makeText(this, "Please fill all form", Toast.LENGTH_SHORT).show();
+        }
+    }
+
```
Comment (zulfiqarlaili) [inline app/src/main/java/com/maybanktest/AddNewItemActivity.java:122]: this part is nice. thanks!

File: app/src/main/java/com/maybanktest/AddNewItemActivity.java [hunk 2]
```diff
@@ -43,95 +41,95 @@ private void ini() {
...
-                    }
-                }else {
-                    if (!TextUtils.isEmpty(txt_title.getText().toString()) && !TextUtils.isEmpty(start_date.getText().toString()) && !TextUtils.isEmpty(end_date.getText().toString())) {
-                        TaskModel taskModel = new TaskModel(getIntent().getExtras().getString("id"), txt_title.getText().toString(), start_date.getText().toString(), end_date.getText().toString(), false);
-                        updateData(taskModel);
-
-                    } else {
-                        Toast.makeText(this, "Please fill all form", Toast.LENGTH_SHORT).show();
-                    }
-                }
```
Comment (zulfiqarlaili) [inline app/src/main/java/com/maybanktest/AddNewItemActivity.java:134]: my part is here is messy. hahaha the if is too long

File: app/src/main/java/com/maybanktest/DatabaseHelper.java [hunk 3]
```diff
@@ -48,56 +47,47 @@ public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
...
 
     }
 
-    public boolean deleteData(String id)
-    {
+    void deleteData(String id) {
         SQLiteDatabase db = this.getWritableDatabase();
-        return db.delete(TABLE_NAME, COLUMN_ID + "=?", new String[]{id}) > 0;
+        db.delete(TABLE_NAME, COLUMN_ID + "=?", new String[]{id});
     }
```
Comment (zulfiqarlaili) [inline app/src/main/java/com/maybanktest/DatabaseHelper.java:86]: i thought i just put return so maybe i can use it later to give indication to user when delete is complete and success. atleast that is my intention.

File: app/src/main/java/com/maybanktest/DatabaseHelper.java [hunk 4]
```diff
@@ -48,56 +47,47 @@ public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
...
-    public Cursor getData(){
+
+    Cursor getData() {
         SQLiteDatabase db = this.getWritableDatabase();
-        String query = "SELECT * FROM "+ TABLE_NAME;
-        Cursor data = db.rawQuery(query,null);
-        return data;
+        String query = "SELECT * FROM " + TABLE_NAME;
+        return db.rawQuery(query, null);
```
Comment (zulfiqarlaili) [inline app/src/main/java/com/maybanktest/DatabaseHelper.java:91]: you really did clean up yea. hahaha

File: app/src/main/java/com/maybanktest/Helper.java [hunk 5]
```diff
@@ -1,41 +1,32 @@
...
-        if (montInNumber == 4) month = "Apr";
-        if (montInNumber == 5) month = "May";
-        if (montInNumber == 6) month = "Jun";
-        if (montInNumber == 7) month = "July";
-        if (montInNumber == 8) month = "Aug";
-        if (montInNumber == 9) month = "Sep";
-        if (montInNumber == 10) month = "Oct";
-        if (montInNumber == 11) month = "Nov";
-        if (montInNumber == 12) month = "Dec";
-        return month;
```
Comment (zulfiqarlaili) [inline app/src/main/java/com/maybanktest/Helper.java:39]: i messed up this one. look so ugly. thank you for your time for this

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1,3 +1,7 @@
 <resources>
     <string name="app_name">MaybankTest</string>
+    <string name="complete">Complete</string>
+    <string name="incomplete">Incomplete</string>
+    <string name="create_now">Create Now</string>
+    <string name="update">Update</string>
```
Comment (zulfiqarlaili) [inline app/src/main/res/values/strings.xml:6]: usually this is my practice to make it easy to change language but im lazy hahaha

**Review by zulfiqarlaili — APPROVED** [review]
i already review the changes. i think ill approve and thanks

File: app/src/main/java/com/maybanktest/DatabaseHelper.java (hunk 3 above)
Comment (yuzurihaaa) [reply app/src/main/java/com/maybanktest/DatabaseHelper.java:86]: see [this](https://martinfowler.com/bliki/Yagni.html) ;)

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (yuzurihaaa) [reply app/src/main/res/values/strings.xml:6]: should not do it lazily in interview test. Take everything as points. You may want them.

**Comment by yuzurihaaa** [comment]
Thank you for your feedback. Good luck!

**Pull request closed without merging**